tests/sibernetic/generate_wcon.py

```diff
@@ -0,0 +1,187 @@
+import math
+import sys
+
+
+def dist(x1, y1, x2, y2):
+    return math.sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2)
+
+
+def generate_wcon(pos_file_name,
+                  wcon_file_name,
+                  rate_to_plot=1,
+                  max_lines=1e12,
+                  max_time_s=1e12,
+                  plot=False):
+
+    postions_file = open(pos_file_name)
+
+    line_num = 0
+
+    print("Loading: %s" % pos_file_name)
+
+    ts = []
+    x = {}
+    y = {}
+    z = {}
+
+    if plot:
+        import matplotlib.pyplot as plt
+        fig = plt.figure()
+        ax = fig.add_subplot(111)
+
+    num_plotted_frames = 0
+    points_plotted = 0
+
+    middle_points = []
+    ave_points = []
+    middle_point_speed_x = []
+    middle_point_speed_y = []
+    ave_point_speed_x = []
+    ave_point_speed_y = []
+    time_points = []
+
+    num_frames = 0
+
+    for line in postions_file:
+
+        line_num += 1
+        if line_num > max_lines:
+            print("Finished parsing file, as max number of lines reached!")
+            break
+
+        if line_num == 1 or line_num % rate_to_plot == 0:
+            num_frames += 1
+            words = line.split()
+            points = (len(words) - 4) / 3
```

Comment:
If you compute this as the number of tokens between X: and Y:, the converter will be far more general.